### ISSUE TITLE
input/number looks strange in FireFox

### ISSUE THREAD

**Issue by jdalsem** [Elgg/Elgg#10288]
![number_input](https://cloud.githubusercontent.com/assets/861833/18865802/569e52a4-849d-11e6-9afb-fd012df1bafe.png)

Don't know if other browser have the same issue

**Comment by iionly** [comment]
I suspect the strange look is something that can't be fixed in Elgg because it's the OS / desktop environment that is handling the look of it. I think it's basically the same issue I had with select input fields (https://github.com/Elgg/Elgg/issues/10276). In my case the look of the select buttons changed / looked right again after changing the GTK theme used in KDE to deal with GTK applications (in this case Firefox). The change of the GTK theme also resulted in a different look of the arrows in the number input fields.

What's the OS / desktop environment (and then maybe also GTK theme) you are using?

**Comment by jdalsem** [comment]
you are right this is a browser related issue... should/can we apply some elgg css to style this cross browser?

**Comment by iionly** [comment]
I think it would be possible to disable the arrows from showing by turning the appearance into a normal textfield with something like:
```
input[type=number] { 
  -moz-appearance: textfield;
}
```
But I don't think it is possible to easily get a look that is consistent (and working) on all browsers and OSes. I tried that for the select inputs when it appeared cut off for me and I gave up at some point because I couldn't just change some attributes without breaking the whole layout and fixing it for Firefox was breaking it for Chrome at one point (only working on Linux so it might even have been broken more for Windows etc. I wasn't even aware of).

**Comment by jdalsem** [comment]
disabling the arrows is something i could get behind.

**Comment by hypeJunction** [comment]
Hence my initial objection to HTML5 input types...

**Comment by iionly** [comment]
> Hence my initial objection to HTML5 input types...

I don't think this is a general issue of HTML5 input types. As I mentioned, I had much the same issue with select input fields just because the desktop theme used was outdated and select inputs are by no way exotic or new. And it wouldn't have been a real solution for the look to be fixed by some Elgg specific CSS as the same issue also showed up on any other website.

Number input fields are already very well supported by all major browsers on all major OSes (as opposed to some other new HTML5 input types). @jdalsem, are you testing on some mainstream OS like Windows or MacOS (where it would really be not a matter of some misconfigured desktop environment) or is it on some Linux desktop enviroment where it is more like an issue of the desktop environment not correctly configured or up to date?

**Comment by iionly** [comment]
Should I make a PR to remove support of number input fields again? I think partial support is worse than no support.

**Comment by mrclay** [comment]
Strawman: Revert and move to master and add a note in docs/admin/upgrading (basically "you can override the view if you don't like this").

**Comment by jdalsem** [comment]
i think we could keep it... it allows easier styling if we start using it more... with the 
```css
input[type=number] { 
  -moz-appearance: textfield;
}
```
there is no real reason to revert

**Comment by mrclay** [comment]
So we just add CSS:
```css
.elgg-input-number {
  -moz-appearance: textfield;
}
```

**Comment by iionly** [comment]
Fixed by https://github.com/Elgg/Elgg/pull/10291.

Though I'm not happy about that as I'm pretty sure that it would have worked perfectly fine for the vast majority of Firefox users.

**Comment by mrclay** [comment]
> Fixed by #10291.

Not until the PR is accepted please. :)
